### ISSUE TITLE
failed to parse documents which have indent heredoc

### ISSUE THREAD

**Issue by skaji** [Perl-Critic/PPI#203]
I expected this test passed, but in fact, it failed.
```perl
use 5.26.0;
use PPI::Document;
use Test::More;

my $content = q(
my $x = <<~EOF;
} /* */
EOF

=head1 NAME

name - abstract

=cut
);

my $doc = PPI::Document->new(\$content);
my $pod = $doc->find('PPI::Token::Pod');

ok $pod, "find pod element";

done_testing;
```
```
❯ prove test.t
test.t .. 1/?
#   Failed test 'find pod element'
#   at test.t line 20.
# Looks like you failed 1 test of 1.
test.t .. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests

Test Summary Report
-------------------
test.t (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=1, Tests=1,  0 wallclock secs ( 0.01 usr  0.01 sys +  0.08 cusr  0.01 csys =  0.11 CPU)
Result: FAIL
```

**Comment by zmughal** [comment]
This can be closed via #209.

**Comment by oalders** [comment]
Thanks for the reminder @zmughal!
